### PR TITLE
chore: use testcontainers-go instead of containertest

### PR DESCRIPTION
**Pull request by mdelapenya**
**Description:** <Describe what has changed.>

This PR is replacing the internal package for managing containers lifecycle in tests with testcontainers-go. Following the same procedure as in #16209, where we updated the Redis receiver tests, we are updating:

- zookeeper receiver
- memcached receiver
- docker stats receiver
- aerospike receiver
- kafka metrics receiver
- docker observer extension

**Link to tracking Issue:**  #9939

**Testing:** <Describe what testing was performed and which tests were added.>

I ran each module tests with:

```shell
cd $the_module
go test -timeout 600s -v -tags integration
```

**Documentation:** No docs were updated

**Follow-ups**
After merging this PR, I think the `containertest` package can be removed/deprecated.

### CODE REVIEW

File: receiver/redisreceiver/go.mod [hunk 1]
```diff
@@ -67,7 +67,5 @@ require (
 
 replace github.com/open-telemetry/opentelemetry-collector-contrib/internal/common => ../../internal/common
 
-replace github.com/open-telemetry/opentelemetry-collector-contrib/internal/containertest => ../../internal/containertest
```
Comment (mdelapenya) [inline receiver/redisreceiver/go.mod:70]: I forgot to remove this line in #16209

**Comment by mdelapenya** [comment]
It seems the same error than yesterday happened at the windows pipeline. Is there anything I can do to troubleshoot it?

**Comment by djaglowski** [comment]
> It seems the same error than yesterday happened at the windows pipeline. Is there anything I can do to troubleshoot it?

I opened https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16267 to track the issue. I don't recall seeing this one in particular before, but we do deal with [flaky tests](https://github.com/open-telemetry/opentelemetry-collector-contrib/labels/flaky%20test) much more frequently than we would like.

Ideally we would resolve every issue but these are often difficult to reproduce/debug. We try to at least document any failures in a corresponding issue and note reoccurrences as well. If nothing else, this provides more information about what may be happening. If a failure occurs frequently we will disable the test until it is resolved. We could certainly use more help with these flaky tests. Any help or ideas for managing are welcome.

**Review by djaglowski — APPROVED** [review]
LGTM. Thanks again @mdelapenya.

**Comment by mdelapenya** [comment]
The error seems a real one in `dockerstatsreceiver.TestRemovedContainerRemovesRecordsIntegration`.

Currently looking at it 👀

**Comment by mdelapenya** [comment]
@djaglowski I did my best trying to understand the test suite 🤞 Result is https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16264/commits/f01b5bda345a765b27828afedabf7726dc289b61

I noticed the test failed because the assertions was expecting an isolated Docker execution, and it could be the case that other containers are running.

I'd recommend checking the current assertions regarding Docker events/stats, as they could be influenced by the existence of, i.e., the Ryuk container or other pending-to-be-pruned containers. Please take a look at https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16264/commits/f01b5bda345a765b27828afedabf7726dc289b61#diff-ca81407f807d8cfe041b4a8ad51b991f51e0fcc20bfd8ff6bad32b9b9ac9e0ddR224 and https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16264/commits/f01b5bda345a765b27828afedabf7726dc289b61#diff-ca81407f807d8cfe041b4a8ad51b991f51e0fcc20bfd8ff6bad32b9b9ac9e0ddR247, where I'm checking the initial count of running containers, and expecting just one more in the assertion.

Same issue in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16264/commits/f01b5bda345a765b27828afedabf7726dc289b61#diff-ca81407f807d8cfe041b4a8ad51b991f51e0fcc20bfd8ff6bad32b9b9ac9e0ddR255, where I created a stub Container from the container ID.

Hopefully I guessed correctly 😅

**Review by antonblock — APPROVED** [review]
Aerospike changes look good and pass integration tests

**Comment by mdelapenya** [comment]
Mongodb receiver tests failing:
```
--- FAIL: TestMongodbIntegration (0.00s)
    --- FAIL: TestMongodbIntegration/Running_mongodb_4.0 (16.87s)
    --- PASS: TestMongodbIntegration/Running_mongodb_4.2 (79.81s)
    --- PASS: TestMongodbIntegration/Running_mongodb_4.4_as_LPU (81.02s)
    --- PASS: TestMongodbIntegration/Running_mongodb_5.0 (76.70s)
FAIL
coverage: 73.3% of statements
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbreceiver	156.561s
?   	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/mongodbreceiver/internal/metadata	[no test files]
FAIL
make[1]: *** [../../Makefile.Common:68: do-integration-tests-with-cover] Error 1
make: *** [Makefile:188: receiver/mongodbreceiver] Error 2
```

We could probably be facing "integrated tests", or tests that could eventually depend on each other's state. Will take a look at it once I have time.

**Comment by mdelapenya** [comment]
I noticed the mongoDB test examples were using a fixed port to be exposed, and that port was the same for mongo [4.0](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16264/commits/ffe8ca5aecb578c513645114b3ba6caa66a008af#diff-3967959f6b8f81572ebce39791ba27a28b2d7e27850d1990861cd219c3fbf736L46) and [4.2](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16264/commits/ffe8ca5aecb578c513645114b3ba6caa66a008af#diff-3967959f6b8f81572ebce39791ba27a28b2d7e27850d1990861cd219c3fbf736L54).

I updated the function to return the mongo container to retrieve both the container and its endpoint, calculating the dynamic, random port internally.

Hope this fixes the build 🤞

**Comment by mdelapenya** [comment]
The build failed because of the fpm install: https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3454696043/jobs/5766407878

```
Run gem install --no-document fpm -v 1.11.0
  gem install --no-document fpm -v 1.11.0
  shell: /usr/bin/bash -e {0}
  env:
    TEST_RESULTS: testbed/tests/results/junit/results.xml
    SEGMENT_DOWNLOAD_TIMEOUT_MINS: 5
ERROR:  While executing gem ... (Gem::RemoteFetcher::FetchError)
    too many connection resets (https://rubygems.org/quick/Marshal.[4](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3454696043/jobs/5766407878#step:4:4).[8](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3454696043/jobs/5766407878#step:4:9)/fpm-1.11.0.gemspec.rz)
Error: Process completed with exit code 1.
```

Probably a retry on the step could help here: https://github.com/orgs/community/discussions/27121#discussioncomment-3254700

**Comment by djaglowski** [comment]
@mdelapenya, really appreciate your efforts on this. 

Given that the mongo changes go beyond what is necessary for the others, would you be willing to open a separate PR for those?

**Comment by mdelapenya** [comment]
> Given that the mongo changes go beyond what is necessary for the others, would you be willing to open a separate PR for that component?

Sure, I can revert the commit for MongoDB and send it in a separate PR. I'm worried if, without the commit, the PR does not get merged because the mongo tests could eventually fail for the inconsistency of the used ports. If you agree with that, then't it's fine for me too :)

**Review by djaglowski — APPROVED** [review]

**Pull request merged**